docs/core/DelegationManager.md

```diff
@@ -0,0 +1,195 @@
+## DelegationManager
+
+| File | Type | Proxy? |
+| -------- | -------- | -------- |
+| [`DelegationManager.sol`](../../src/contracts/core/DelegationManager.sol) | Singleton | Transparent proxy |
+
+The primary functions of the `DelegationManager` are (i) to allow Stakers to delegate to Operators and (ii) to keep an up-to-date record of the number of shares each Operator has been delegated for each strategy.
+
+Whereas the `EigenPodManager` and `StrategyManager` perform accounting for individual Stakers according to their native ETH or LST holdings respectively, the `DelegationManager` sits between these two contracts and tracks these accounting changes according to the Operators each Staker has delegated to. This means that each time a Staker's balance changes in either the `EigenPodManager` or `StrategyManager`, the `DelegationManager` is called to record this update to the Staker's delegated Operator (if they have one).
+
+*Does not include*:
+* Pausability ("Pausers" section) or Upgradability
+* High-level flows (what high-level flows this contract has entry points for)
+
+### Operators
+
+Operators interact with the following functions:
+
+#### `registerAsOperator`
+
+```solidity
+function registerAsOperator(OperatorDetails calldata registeringOperatorDetails, string calldata metadataURI) external
+```
+
+Registers the caller as an Operator in EigenLayer. The new Operator provides the `OperatorDetails`, a struct containing:
+* `address earningsReceiver`: the address that will receive earnings as the Operator provides services to AVSs *(currently unused)*
+* `address delegationApprover`: if set, this address must sign and approve new delegation from Stakers to this Operator *(optional)*
+* `uint32 stakerOptOutWindowBlocks`: the minimum delay (in blocks) between beginning and completing registration for an AVS. *(currently unused)*
+
+`registerAsOperator` cements the Operator's `OperatorDetails`, and self-delegates the Operator to themselves - permanently marking the caller as an Operator. They cannot "deregister" as an Operator - however, they can exit the system by withdrawing their funds via the `EigenPodManager` or `StrategyManager`.
+
+**Effects**:
+* Sets `OperatorDetails` for the Operator in question
+* Delegates the Operator to itself
+* If the Operator has deposited into the `EigenPodManager` and is not in undelegation limbo, the `DelegationManager` adds these shares to the Operator's shares for the beacon chain ETH strategy.
+* For each of the three strategies in the `StrategyManager`, if the Operator holds shares in that strategy they are added to the Operator's shares under the corresponding strategy.
+
+**Requirements**:
+* Caller MUST NOT already be an Operator
+* Caller MUST NOT already be delegated to an Operator
+* `earningsReceiver != address(0)`
+* `stakerOptOutWindowBlocks <= MAX_STAKER_OPT_OUT_WINDOW_BLOCKS`: (~15 days)
```

Comment:
this is actually ~180 days. Looks like it's being calculated as `(180 days) / 12` to calculate the  ~number of blocks in 180 days, so the confusion is understandable